File: backend-java/minimarket-api-java/src/main/java/com/minimarket/api/controller/ReceiptsController.java
package com.minimarket.api.controller;

import com.minimarket.api.dto.ApiMessageResponse;
import com.minimarket.api.dto.ReceiptConfirmRequestDto;
import com.minimarket.api.dto.ReceiptScanRequestDto;
import com.minimarket.api.service.ReceiptService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/receipts")
public class ReceiptsController {

    private final ReceiptService receiptService;

    public ReceiptsController(ReceiptService receiptService) {
        this.receiptService = receiptService;
    }

    // POST /api/receipts/scan -> lee la foto de la boleta con IA y propone productos/costos.
    @PostMapping("/scan")
    public ResponseEntity<?> scan(@RequestBody ReceiptScanRequestDto request) {
        var result = receiptService.scan(request);
        if (!result.success()) {
            return ResponseEntity.badRequest().body(new ApiMessageResponse(result.error()));
        }
        return ResponseEntity.ok(result.data());
    }

    // POST /api/receipts/confirm -> guarda los productos/costos confirmados por el usuario.
    @PostMapping("/confirm")
    public ResponseEntity<?> confirm(@RequestBody ReceiptConfirmRequestDto request) {
        var result = receiptService.confirm(request);
        if (!result.success()) {
            return ResponseEntity.badRequest().body(new ApiMessageResponse(result.error()));
        }
        return ResponseEntity.ok(result.data());
    }
}

File: backend/Minimarket.Api/Controllers/ReceiptsController.cs
using Microsoft.AspNetCore.Mvc;
using Minimarket.Api.DTOs;
using Minimarket.Api.Services;

namespace Minimarket.Api.Controllers;

[ApiController]
[Route("api/receipts")]
public class ReceiptsController(IReceiptService receiptService) : ControllerBase
{
    // POST api/receipts/scan -> lee la foto de la boleta con IA y propone productos/costos.
    [HttpPost("scan")]
    public async Task<ActionResult<ReceiptScanResultDto>> Scan([FromBody] ReceiptScanRequestDto request)
    {
        var result = await receiptService.ScanAsync(request);
        if (!result.Success)
        {
            return BadRequest(new { message = result.Error });
        }

        return Ok(result.Result);
    }

    // POST api/receipts/confirm -> guarda los productos/costos confirmados por el usuario.
    [HttpPost("confirm")]
    public async Task<ActionResult<ReceiptConfirmResultDto>> Confirm([FromBody] ReceiptConfirmRequestDto request)
    {
        var result = await receiptService.ConfirmAsync(request);
        if (!result.Success)
        {
            return BadRequest(new { message = result.Error });
        }

        return Ok(result.Result);
    }
}
